src/Microsoft.PowerShell.Commands.Utility/commands/utility/join-string.cs

```diff
@@ -0,0 +1,210 @@
+        private bool _firstInputObject = true;
+
+        /// <summary>
+        /// Gets or sets the property name or script block to use as the value to join.
+        /// </summary>
+        [Parameter(Position = 0)]
+        [ArgumentCompleter(typeof(PropertyNameCompleter))]
+        public PSPropertyExpression Property { get; set; }
+
+        /// <summary>
+        /// Gets or sets the delimiter to join the output with.
+        /// </summary>
+        [Parameter(Position = 1)]
+        [ArgumentCompleter(typeof(JoinItemCompleter))]
+        [AllowEmptyString]
+        public string Separator
+        {
+            get => _separator ?? LanguagePrimitives.ConvertTo<string>(GetVariableValue("OFS"));
+            set => _separator = value;
+        }
+
+        /// <summary>
+        /// Gets or sets text to include before the joined input text.
+        /// </summary>
+        [Parameter]
+        [Alias("op")]
+        public string OutputPrefix { get; set; }
+
+        /// <summary>
+        /// Gets or sets text to include after the joined input text.
+        /// </summary>
+        [Parameter]
+        [Alias("os")]
+        public string OutputSuffix { get; set; }
+
+        /// <summary>
+        /// Gets or sets if the output items should we wrapped in single quotes.
+        /// </summary>
+        [Parameter(ParameterSetName = "SingleQuote")]
+        public SwitchParameter SingleQuote { get; set; }
+
+        /// <summary>
+        /// Gets or sets if the output items should we wrapped in double quotes.
+        /// </summary>
+        [Parameter(ParameterSetName = "DoubleQuote")]
+        public SwitchParameter DoubleQuote { get; set; }
+
+        /// <summary>
+        /// Gets or sets a format string that is applied to each input object.
+        /// </summary>
+        [Parameter(ParameterSetName = "Format")]
+        [ArgumentCompleter(typeof(JoinItemCompleter))]
+        public string FormatString { get; set; }
+
+        /// <summary>
+        /// Gets or sets the input object to join into text.
+        /// </summary>
+        [Parameter(ValueFromPipeline = true)]
+        public PSObject InputObject { get; set; }
+
+        /// <inheritdoc />
+        protected override void BeginProcessing()
+        {
+            _quoteChar = SingleQuote ? '\'' : DoubleQuote ? '"' : char.MinValue;
+            _outputBuilder.Append(OutputPrefix);
+        }
+
+        /// <inheritdoc />
+        protected override void ProcessRecord()
+        {
+            if (InputObject != null && InputObject != AutomationNull.Value)
+            {
+                var inputValue = Property == null
+                                    ? InputObject
+                                    : Property.GetValues(InputObject, false, true).FirstOrDefault()?.Result;
+                var stringValue = LanguagePrimitives.ConvertTo<string>(inputValue);
+
+                if (_firstInputObject)
+                {
+                    _firstInputObject = false;
```

Comment:
Now we do not need this

Reply:
Well, we still need a way to determine if we should skip the first delimiter, don't we?

Reply:
Yes, I forget about the separator.